container.go

```diff
@@ -348,3 +348,17 @@ func getPsEnvVars(checkpointOutputDir string, pid uint32) (envVars []string, err
 
 	return
 }
+
+// findPsTreeByPID performs a short-circuiting depth-first search to find
+// a process with a given PID in a process tree.
+func findPsTreeByPID(root *crit.PsTree, targetPID uint32) *crit.PsTree {
```

Comment:
Would it be more appropriate to name this `findPs`? . It is well known that a PID is the primary identifier for a process, and we are searching for a process rather than a process tree.

Reply:
It might also make sense to provide this as a method on `*PsTree` through go-criu, so that it can be used like `tree.FindPs(<pid>)`.

Reply:
> It might also make sense to provide this as a method on `*PsTree` through go-criu,

This make sense to me as well. Can I add this  as `PsTree` method inside `go-criu/crit`  @rst0git , @adrianreber  ?

Reply:
Indeed, depth-first search for PsTree has many possible use cases. It would make sense to be in go-criu.